feat: Added support for delegation sets

flora-five opened the pull request:
## Description
Add delegation_sets module for creating delegation sets. The implementation (logic, variables and outputs) is similar to the implementation of the zones module.

A delegation set id may be specified in the parameters of a public zone to have that zone use the set of name servers from the delegation set.

Update the example to create and use a delegation set.

## Motivation and Context
By supporting the creation and use of delegation sets, multiple public zones can use the same set of name servers. May be useful when doing batch updates of name servers at other registrars.

## Breaking Changes
It shouldn't break anything, new functionality is added.

## How Has This Been Tested?
Tested by deploying the example.

shiron-babi commented:
@flora-five - Thanks for this. May be very useful. Is it possible to fix the conflicts? 

@antonbabenko - any chance to promote this?

flora-five commented:
@shiron-babi Thanks for the interest!

I have updated the code, the documentation and, instead of modifying the complete example, I've added a simple usage example in the module README.

antonbabenko commented:
This PR looks indeed useful. I apologize that it took me so long to acknowledge that :)

Could you also update the code in `examples/complete` for one zone to show it in action?

flora-five commented:
Thank you, no problem :)

Yes, I would like to do that. If I understand correctly, I should modify one existing zone to use a delegation set, am I right?

antonbabenko commented:
Yes, and make this set using the new submodule.

flora-five commented:
Done, I hope that it is ok.

antonbabenko commented:
Thank you!

antonbabenko commented:
This PR is included in version 2.5.0 :tada:
